docs/src/test-pom.md

```diff
@@ -0,0 +1,329 @@
+    this.page = page;
+    this.getStartedLink = page.locator('a', { hasText: 'Get started' });
+    this.gettingStartedHeader = page.locator('h1', { hasText: 'Getting started' });
+    this.pomLink = page.locator('li', { hasText: 'Playwright Test' }).locator('a', { hasText: 'Page Object Model' });
+    this.tocList = page.locator('article div.markdown ul > li > a');
+  }
+
+  async goto() {
+    await this.page.goto('https://playwright.dev');
+  }
+
+  async getStarted() {
+    await this.getStartedLink.first().click();
+    await expect(this.gettingStartedHeader).toBeVisible();
+  }
+
+  async pageObjectModel() {
+    await this.getStarted();
+    await this.pomLink.click();
+  }
+}
+```
+
+Now we can use the `PlaywrightDevPage` class in our tests.
+
+```js tab=js-js
+// example.spec.js
+const { test, expect } = require('@playwright/test');
+const { PlaywrightDevPage } = require('./playwright-dev-page');
+
+test('getting started should contain table of contents', async ({ page }) => {
+  const playwrightDev = new PlaywrightDevPage(page);
+  await playwrightDev.goto();
+  await playwrightDev.getStarted();
+  await expect(playwrightDev.tocList).toHaveText([
+    'Installation',
+    'First test',
+    'Configuration file',
+    'Writing assertions',
+    'Using test fixtures',
+    'Using test hooks',
+    'VS Code extension',
+    'Command line',
+    'Configure NPM scripts',
+    'Release notes'
+  ]);
+});
+
+test('should show Page Object Model article', async ({ page }) => {
+  const playwrightDev = new PlaywrightDevPage(page);
+  await playwrightDev.goto();
+  await playwrightDev.pageObjectModel();
+  await expect(page.locator('article')).toContainText('Page Object Model is a common pattern');
+});
+```
+
+```js tab=js-ts
+// example.spec.ts
+import { test, expect } from '@playwright/test';
+import { PlaywrightDevPage } from './playwright-dev-page';
+
+test('getting started should contain table of contents', async ({ page }) => {
+  const playwrightDev = new PlaywrightDevPage(page);
+  await playwrightDev.goto();
+  await playwrightDev.getStarted();
+  await expect(playwrightDev.tocList).toHaveText([
+    'Installation',
+    'First test',
+    'Configuration file',
+    'Writing assertions',
+    'Using test fixtures',
+    'Using test hooks',
+    'VS Code extension',
+    'Command line',
+    'Configure NPM scripts',
+    'Release notes'
+  ]);
+});
+
+test('should show Page Object Model article', async ({ page }) => {
+  const playwrightDev = new PlaywrightDevPage(page);
+  await playwrightDev.goto();
+  await playwrightDev.pageObjectModel();
+  await expect(page.locator('article')).toContainText('Page Object Model is a common pattern');
+});
+```
+
+
+
+## Implementation
+###### Playwright Test vs. Library
+* langs: js
+
+:::caution
+This guide is for [Playwright Library](./library.md), if you are using Playwright Test (`@playwright/test`) see [here](./test-pom.md).
+:::
+
+Page object models wrap over a Playwright [Page].
+
+```js
```

Comment:
we can drop the JS example there, since we have it based on the heading in the beginning.